aptos-move/move-examples/token_objects/ambassador/move/sources/ambassador.move

```diff
@@ -0,0 +1,220 @@
+module ambassador_token::ambassador {
+    use std::error;
+    use std::option;
+    use std::string::{String};
+    use std::signer;
+
+    use aptos_framework::object::{Self, Object};
+    use aptos_token_objects::collection;
+    use aptos_token_objects::token;
+
+    // The token does not exist
+    const ETOKEN_DOES_NOT_EXIST: u64 = 1;
+    /// The provided signer is not the creator
+    const ENOT_CREATOR: u64 = 2;
+    /// Attempted to mutate an immutable field
+    const EFIELD_NOT_MUTABLE: u64 = 3;
+    /// Attempted to burn a non-burnable token
+    const ETOKEN_NOT_BURNABLE: u64 = 4;
+    /// Attempted to mutate a property map that is not mutable
+    const EPROPERTIES_NOT_MUTABLE: u64 = 5;
+    // The collection does not exist
+    const ECOLLECTION_DOES_NOT_EXIST: u64 = 6;
+
+    struct AmbassadorCollection has key {
+        /// Used to mutate collection fields
+        mutator_ref: collection::MutatorRef,
+    }
+
+    struct AmbassadorToken has key {
+        /// Used to burn.
+        burn_ref: token::BurnRef,
+        /// Used to control freeze.
+        transfer_ref: object::TransferRef,
+        /// Used to mutate fields
+        mutator_ref: token::MutatorRef,
+    }
+
+    struct AmbassadorLevel has key {
```

Comment:
I'd take this chance and talk to @bowenyang007 as well regarding well-defined attributes like this. On one hand, I really like this as it brings typing and structure into the token model. On the other hand, @bowenyang007 has some concerns on supporting this due to the dynamic nature

Reply:
While keeping this resource as it is, for the comparison sake, I added another attribute called "rank" as a property in the property map.

Reply:
I revised the module to update "Rank" (e.g., "Bronze", "Silver", "Gold") based on the current level.